modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -931,7 +931,11 @@ def apply_idx_objs(df, idx):
                                 slice(cum_row_lengths[i], cum_row_lengths[i + 1])
                             ],
                             length=self.row_lengths[i],
-                            width=self.column_widths[j],
+                            width=(
+                                self.column_widths[j]
+                                if self._column_widths_cache is not None
```

Comment:
Do you mind adding `copy_column_widths_cache` function similar to `copy_columns_cache`?

Reply:
This would require changing `self._column_widths_cache` -> `self.copy_column_widths_cache()` everywhere. Adding the method also doesn't seem to help here, as we're taking individual elements, rather than the whole list:
```python
part.add_to_apply_calls(
    ...,
    width=self.copy_column_widths_cache()[i] if self.has_column_widths_cache() else None
)
```